eth/downloader/statesync.go

```diff
@@ -245,15 +274,24 @@ func (s *stateSync) Cancel() error {
 // receive data from peers, rather those are buffered up in the downloader and
 // pushed here async. The reason is to decouple processing from data receipt
 // and timeouts.
-func (s *stateSync) loop() error {
+func (s *stateSync) loop() (err error) {
 	// Listen for new peer events to assign tasks to them
 	newPeer := make(chan *peer, 1024)
+	peerSub := s.d.peers.SubscribeNewPeers(newPeer)
+	defer peerSub.Unsubscribe()
+	defer func() {
+		cerr := s.commit(true)
+		if err == nil {
```

Comment:
Not sure I love this, b/c an error can be silently dropped.

👓 PTAL.

Reply:
And there are a lot of different kind of errors that it could be, including a commit error. Which seems important.